compiler/rustc_resolve/src/late.rs

```diff
@@ -693,6 +700,9 @@ struct LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
 
     /// Count the number of places a lifetime is used.
     lifetime_uses: FxHashMap<LocalDefId, LifetimeUseSet>,
+
+    /// Record poly-trait-ref, only used for diagnostic.
+    with_poly_trait_ref: FxHashMap<NodeId, RedundantPolyTraitRef>,
```

Comment:
Don't know if this is possible, I'm on mobile rn and don't remember the details of name resolution off the top of my head but instead of equipping the late resolution visitor with extra data can't we just look at the preexisting `lifetime_ribs` and find the innermost `LifetimeRib` whose kind is `LifetimeRibKind::Generics { kind: LifetimeBinderKind::WhereClause, ..}`? I'm assuming here that `WhereClause` refers to the “outer binder” / the binder arching over the entire bound (whereas `LifetimeBinderKind::PolyTrait` refers to the “inner binder” which isn't what we want as outlined in the issue).